src/integration.test.ts

```diff
@@ -418,6 +419,72 @@ describe(`Integration test for portal ${portal}`, () => {
       expect(metadata).toEqual(expect.objectContaining({ filename: customFilename }));
     });
 
+    it("Should upload and download two files with different names and compare their etags", async () => {
```

Comment:
etags will be different only when 
- the skylink is different
- the served path is different
- the format is different

I don't think this is reflected yet in the test suite here.

I think a good test for instance would be to upload a directory with `index.html` and then fetch the skylink at root or at `index.html`... those two downloads should give you the same etag because the first one will have defaulted to `index.html` because of the default path feature.

Another good test would be to download a skyfile through a v1 and v2 skylink and verify both return the same etag. Then lastly we should be downloading a file using a certain format (zip for instance) and verify it returns a different etag...

Reply:
Is skynet-js the place for all these tests? Does skyd not have etag tests? Can be a followup if needed.

Reply:
We do have tests for this in `skyd`, but something slipped through. I think `skynet-js` is definitely a good place to have another line of defense against regression. Often we extend the functionality without considering etags... it's easy to forget. This PR is fully catered to adding etag tests so I wouldn't follow up but add them here.

Reply:
`servedPath` being in the etag for instance is not covered by the current tests

Reply:
etags are not exposed by `skynet-js`, etags are a concern for the server and the browser, but not the client application. I had to do some minor hacks to implement these tests. That to me signifies that this is not the place to be testing this feature. I'm leaving these tests as a sanity check but I do not think that extensive testing of etags here is appropriate.

Reply:
@kwypchlo specifically had to add `etag` to the access control headers in order for these tests to work, which makes me further think that client applications are not really intended to have access to them.